internal/controller/util/json_util.go

```diff
@@ -0,0 +1,348 @@
+		Name:      hook.NameSelector,
+	}
+	switch hook.SelectResource {
+	case "pod":
+		// handle pod type
+		resource := &corev1.Pod{}
+		err := WaitUntilResourceExists(client, resource, nsScopedName, time.Duration(timeout)*time.Second)
+		if err != nil {
+			return false, err
+		}
+		podBytes, err := json.Marshal(resource)
+		if err != nil {
+			return false, err
+		}
+		jsonData := make(map[string]interface{})
+		err = json.Unmarshal(podBytes, &jsonData)
+		if err != nil {
+			return false, err
+		}
+		return evaluateCheckHookExp(hook.Chk.Condition, resource)
+	case "deployment":
+		// handle deployment type
+		resource := &appsv1.Deployment{}
+		err := WaitUntilResourceExists(client, resource, nsScopedName, time.Duration(timeout)*time.Second)
+		if err != nil {
+			return false, err
+		}
+		depBytes, err := json.Marshal(resource)
+		if err != nil {
+			return false, err
+		}
+		jsonData := make(map[string]interface{})
+		err = json.Unmarshal(depBytes, &jsonData)
+		if err != nil {
+			return false, err
+		}
+		return evaluateCheckHookExp(hook.Chk.Condition, resource)
+	case "statefulset":
+		// handle statefulset type
+		resource := &appsv1.StatefulSet{}
+		err := WaitUntilResourceExists(client, resource, nsScopedName, time.Duration(timeout)*time.Second)
+		if err != nil {
+			return false, err
+		}
+		statefulSetBytes, err := json.Marshal(resource)
+		if err != nil {
+			return false, err
+		}
+		jsonData := make(map[string]interface{})
+		err = json.Unmarshal(statefulSetBytes, &jsonData)
+		if err != nil {
+			return false, err
+		}
+		return evaluateCheckHookExp(hook.Chk.Condition, resource)
+	}
+
+	return false, nil
+}
+
+func WaitUntilResourceExists(client client.Client, obj client.Object, nsScopedName types.NamespacedName,
+	timeout time.Duration,
+) error {
+	ctx, cancel := context.WithTimeout(context.Background(), timeout)
+	defer cancel()
+
+	ticker := time.NewTicker(100 * time.Millisecond) // Poll every 100 milliseconds
+	defer ticker.Stop()
+
+	for {
+		select {
+		case <-ctx.Done():
+			return fmt.Errorf("timeout waiting for resource %s to be ready: %w", nsScopedName.Name, ctx.Err())
+		case <-ticker.C:
+			err := client.Get(context.Background(), nsScopedName, obj)
+			if err != nil {
+				if k8serrors.IsNotFound(err) {
+					// Resource not found, continue polling
+					continue
+				}
+
+				return err // Some other error occurred, return it
+			}
+
+			return nil // Resource is ready
+		}
+	}
+}
+
+func getTimeoutValue(hook *kubeobjects.HookSpec) int {
+	if hook.Chk.Timeout != 0 {
+		return hook.Chk.Timeout
+	} else if hook.Timeout != 0 {
+		return hook.Timeout
+	} else {
+		// 300s is the default value for timeout
+		return 300
+	}
+}
+
+func evaluateCheckHookExp(booleanExpression string, jsonData interface{}) (bool, error) {
+	op, jsonPaths, err := parseBooleanExpression(booleanExpression)
+	if err != nil {
+		return false, fmt.Errorf("failed to parse boolean expression: %w", err)
+	}
+
+	operand := make([]reflect.Value, 2)
```

Comment:
This line creates a slice with exactly two slots, which means it only accommodates two jsonPaths. If jsonPaths contains more than two elements, the next loop will try to access operand[i] with i greater than 1, leading to an index out of range error.